_posts/2020-10-26-dynamically-changeable-logging.md

```diff
@@ -0,0 +1,150 @@
+
+For Kafka brokers:
+
+```
+$ kubectl  exec <kafka-cluster-name>-zookeeper-0 -- bin/kafka-configs.sh --bootstrap-server <kafka-cluster-name>-kafka-bootstrap:9092 --entity-type broker-loggers --entity-name 0 --describe
+```
+
+For Kafka Connect you can get loggers information by accessing Connect API service:
+
+```
+$ kubectl exec -ti <kafka-cluster-name>-kafka-0 -- curl http://<connect-cluster-name>-connect-api:8083/admin/loggers
+```
+
+Since Kafka MirrorMaker2 is based on the Kafka Connect, getting its loggers is very similar:
+```
+$ kubectl exec -ti <kafka-cluster-name>-kafka-0 -- curl http://<mm2-cluster-name>-connect-api:8083/admin/loggers
+```
+
+Strimzi Cluster Operator, Topic Operator, Entity Operator and Strimzi Kafka Bridge use automatic reloading of logging configuration.
+The only way to check the logging configuration of these components is by looking into the pod for the file on the path `/opt/strimzi/custom-config/log4j2.properties`.
+It should contain the configuration and the `monitorInterval` property.
+Note that reconciliation time and `monitorInterval` value affect the time logging configuration to be reloaded.
+
+## Implementation details
+
+To implement dynamically changeable logging levels in Strimzi Cluster Operator, Topic Operator, Entity Operator and Strimzi Kafka Bridge we use the`log4j2` [automatic reconfiguration](https://logging.apache.org/log4j/log4j-2.1/manual/configuration.html#AutomaticReconfiguration) feature.
+
+Automatic reloading works in these steps:
+1. Logging configuration is changed in the custom resource
+2. During reconciliation the `log4j2.properties`  file is remounted. It contains the configuration from the custom resource.
+3. `log4j2` polls changes in the `log4j2.properties` file in an interval of `monitorInterval` seconds
+
+For Kafka broker, Kafka Connect and Kafka MirrorMaker2 we use features already implemented from Kafka.
+Logging configuration in Kafka Brokers is changed using `AdminClient`.
+For more information, see [KIP-412](https://cwiki.apache.org/confluence/display/KAFKA/KIP-412%3A+Extend+Admin+API+to+support+dynamic+application+log+levels)
+
+Kafka Connect and Kafka MirrorMaker2 logging configuration is changed using the REST API.
+For more information, see [KIP-495](https://cwiki.apache.org/confluence/display/KAFKA/KIP-495%3A+Dynamically+Adjust+Log+Levels+in+Connect)
+
+## Further steps
+
+For now, there is no way to set logging configuration for Kafka MirrorMaker.
+There is a [KIP-649](https://cwiki.apache.org/confluence/display/KAFKA/KIP-649%3A+Dynamic+Client+Configuration) in progress.
+However, Kafka MirrorMaker support will eventually be dropped, as it has been replaced by MirrorMaker 2.0.. 
+
+
+An upgrade to `log4j` has been proposed: [KIP-653](https://cwiki.apache.org/confluence/display/KAFKA/KIP-653%3A+Upgrade+log4j+to+log4j2) arised.
+For now, it is not clear how this upgrade will change dynamic changes to logging levels.
```

Comment:
I would maybe leave this out as it might give a feeling of uncertainty. I would just leave the link for the KIP and maybe say something like that will make it easier (because of the minotorInterval or something).